packages/desktop-client/src/hooks/useCategoryScheduleGoalTemplateIndicator.ts

```diff
@@ -0,0 +1,142 @@
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { type TFunction } from 'i18next';
+
+import { type ScheduleStatusType } from 'loot-core/client/data-hooks/schedules';
+import * as monthUtils from 'loot-core/shared/months';
+import {
+  type CategoryEntity,
+  type ScheduleEntity,
+} from 'loot-core/types/models';
+
+import { useCategoryScheduleGoalTemplates } from './useCategoryScheduleGoalTemplates';
+import { useLocale } from './useLocale';
+
+type UseCategoryScheduleGoalTemplateProps = {
+  category: CategoryEntity;
+  month: string;
+};
+
+type UseCategoryScheduleGoalTemplateResult = {
+  schedule: ScheduleEntity | null;
+  scheduleStatus: ScheduleStatusType | null;
+  isScheduleRecurring: boolean;
+  description: string;
+};
+
+/**
+ * We only display an indicator for one schedule at a time, so we
+ * filter and then sort the schedules to show missed schedules first,
+ * then due schedules, then upcoming schedules. This is to ensure that
+ * the user sees the most important schedule first.
+ */
+export function useCategoryScheduleGoalTemplateIndicator({
+  category,
+  month,
+}: UseCategoryScheduleGoalTemplateProps): UseCategoryScheduleGoalTemplateResult {
+  const { t } = useTranslation();
+  const locale = useLocale();
+
+  const { schedules, statuses: scheduleStatuses } =
+    useCategoryScheduleGoalTemplates({
+      category,
+    });
+
+  return useMemo<UseCategoryScheduleGoalTemplateResult>(() => {
+    const schedulesToDisplay = schedules
+      .filter(schedule => {
+        const status = scheduleStatuses.get(schedule.id);
+        return status === 'upcoming' || status === 'due' || status === 'missed';
+      })
+      .filter(
+        schedule => monthUtils.monthFromDate(schedule.next_date) === month,
+      )
+      .sort((a, b) => {
+        // Display missed schedules first, then due, then upcoming.
+        const aStatus = scheduleStatuses.get(a.id);
+        const bStatus = scheduleStatuses.get(b.id);
+        if (aStatus === 'missed' && bStatus !== 'missed') return -1;
+        if (bStatus === 'missed' && aStatus !== 'missed') return 1;
+        if (aStatus === 'due' && bStatus !== 'due') return -1;
+        if (bStatus === 'due' && aStatus !== 'due') return 1;
+        return 0;
+      });
+
+    const description = schedulesToDisplay
+      .map(s => {
+        return getScheduleStatusDescription({
+          t,
+          schedule: s,
+          scheduleStatus: scheduleStatuses.get(s.id),
+          locale,
+        });
+      })
+      .reduce((acc, tooltip) => {
+        return `${acc}\n${tooltip}`;
+      }, '');
```

Comment:
I'm getting an extra blank line at the top of the tooltips because the first iteration of this loop adds '\n' before the first entry.

```suggestion
      .join('\n');
```